src/app/code/community/IntegerNet/Solr/Block/Result/List.php

```diff
@@ -76,10 +76,17 @@ protected function _beforeToHtml()
         // called prepare sortable parameters
         $collection = $this->_getProductCollection();
 
-        // use sortable parameters
-        if ($orders = $this->getAvailableOrders()) {
+        // use sortable parameters depending on category settings if available
+        if (Mage::helper('integernet_solr')->page()->isCategoryPage()
+            && $category = Mage::registry('current_category')) {
+            $orders = $category->getAvailableSortByOptions();
```

Comment:
Now that I see this, `sortByOptions` is definitely a better name choice than `orders` :)

Reply:
Funny that both are core methods